crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -57,7 +57,56 @@ def prepare
   api_version "2.0"
   def services
     if request.post?
-      head :not_implemented
+
+      # prepare the scripts for various actions in one chef run
+      service_object = CrowbarService.new(Rails.logger)
+      service_object.prepare_nodes_for_os_upgrade
```

Comment:
can this fail somehow, and do we need to report/handle a failure here for the caller of the api?